kornia/augmentation/_2d/geometric/resize.py

```diff
@@ -114,6 +114,12 @@ def __init__(
             p=p,
         )
 
+    def apply_transform(
+        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
+    ) -> Tensor:
+        self._param_generator = rg.ResizeGenerator(resize_to=self.flags['size'], side=self.flags['side'])
```

Comment:
```suggestion
        self._param_generator.output_size = self.flags['size']
```

I don't know what makes more sense here, creating another ParamGenerator for these cases, or update of the resize size as this suggestion, or some other thing

for default, the `ResizeGenerator` set the `output_size` on the init (https://github.com/kornia/kornia/blob/master/kornia/augmentation/random_generator/_2d/resize.py#L34), and then when expands it from int to a shape (https://github.com/kornia/kornia/blob/master/kornia/augmentation/random_generator/_2d/resize.py#L69)

as I understand it, for these MaxSize cases we need to update this size on each call, because the `output_size` depends on the input size (which side is bigger)

@shijianjian